tests/foreman/endtoend/test_api_endtoend.py

```diff
@@ -1185,13 +1184,16 @@ def test_positive_end_to_end(self, function_sca_manifest, target_sat, rhel_conte
         # step 2.18: Provision a client
         # TODO this isn't provisioning through satellite as intended
         # Note it wasn't well before the change that added this todo
-        # Register client with foreman server using act keys
-        rhel_contenthost.register(org, None, activation_key.name, target_sat)
+        # Register client with foreman server using act keys and install packages
+        packages = 'katello-host-tools puppet-agent'
+        result = rhel_contenthost.api_register(
```

Comment:
In the OP, you say: `API e2e test to use hammer to register instead of API/nailgun`. So which is it and why?

Reply:
register helper uses hammer to register a host, whereas api_register helper uses API

Reply:
Yes. Here, you change from Hammer to API. Meanwhile, OP states the opposite.

Reply:
I think its a problem statement for this PR, and currently it uses hammer which would be changed to API as solution, right?